src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -2834,7 +2834,7 @@ public function get_modifiable_text() {
 			return '';
 		}
 
-		$text = substr( $this->html, $this->text_starts_at, $this->text_length );
+		$text = str_replace( "\0", '', substr( $this->html, $this->text_starts_at, $this->text_length ) );
```

Comment:
It's not as simple as this.

Depending on the context, the bytes may be ignored or they may be replaced with `�`.